src/Compilers/CSharp/Test/Emit2/Semantics/PatternMatchingTests.cs

```diff
@@ -11784,5 +11786,80 @@ public void Test2(G<object> x2)
                 Diagnostic(ErrorCode.ERR_BadBinaryOps, "x2 as T").WithArguments("as", "G<object>", "T").WithLocation(11, 18)
                 );
         }
+
+        [Fact]
+        [WorkItem(63476, "https://github.com/dotnet/roslyn/issues/63476")]
+        public void PatternNonConstant_UserDefinedImplicit_ConvertionToInputType()
+        {
+            var source =
+@"
+class A {
+    public string S { get; set; }
+    public static implicit operator A(string s) { return new A { S = s }; }
+}
+class C
+{
+    static bool M(A a) => a switch { ""implicitA"" => true, _ => false };
+}";
+            CreateCompilationWithSpanAndMemoryExtensions(source, parseOptions: TestOptions.RegularPreview)
+                .VerifyDiagnostics(
+                    // (8,38): error CS9098: Cannot convert type 'string' to type 'A' using user-defined conversion in pattern expression
```

Comment:
We should also test a subpattern case like `a switch { "implicitA" or { S: { } } => true, ... }`.